### PR TITLE
python3Packages.pyannote-audio: init at 3.1.1 (+16 Dependencies)

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Description of changes

Adds pyannote-audio to Nixpkgs, a dependency of a lot of text to speech projects to Nixpkgs, including its dependency chain of 16 other packages.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pyannote-audio/default.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  ];
+
+  pythonImportsCheck = [ "pyannote.audio" ];
+
+  meta = with lib; {
+    description = "Neural building blocks for speaker diarization: speech activity detection, speaker change detection, overlapped speech detection, speaker embedding";
+    homepage = "https://github.com/pyannote/pyannote-audio";
+    changelog = "https://github.com/pyannote/pyannote-audio/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (ehmry) [inline pkgs/development/python-modules/pyannote-audio/default.nix:85]: Empty maintainers.

File: pkgs/development/python-modules/pyannote-core/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    wheel
+  ];
+
+  pythonImportsCheck = [ "pyannote.core" ];
+
+  meta = with lib; {
+    description = "Advanced data structures for handling temporal segments with attached labels";
+    homepage = "https://github.com/pyannote/pyannote-core";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (ehmry) [inline pkgs/development/python-modules/pyannote-core/default.nix:42]: Empty maintainers.

**Review by ghost — CHANGES_REQUESTED** [review]
Is there a reason to use `fetchFromGithub` in some places and not use `fetchPypi` everywhere?

File: pkgs/development/python-modules/pyannote-metrics/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    wheel
+  ];
+
+  pythonImportsCheck = [ "pyannote.metrics" ];
+
+  meta = with lib; {
+    description = "A toolkit for reproducible evaluation, diagnostic, and error analysis of speaker diarization systems";
+    homepage = "https://github.com/pyannote/pyannote-metrics";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (ehmry) [inline pkgs/development/python-modules/pyannote-metrics/default.nix:54]: Empty maintainers.

File: pkgs/development/python-modules/pyannote-pipeline/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    wheel
+  ];
+
+  pythonImportsCheck = [ "pyannote.pipeline" ];
+
+  meta = with lib; {
+    description = "Tunable pipelines";
+    homepage = "https://github.com/pyannote/pyannote-pipeline";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (ehmry) [inline pkgs/development/python-modules/pyannote-pipeline/default.nix:50]: Empty maintainers.

File: pkgs/development/python-modules/pyscaffoldext-django/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    ];
+  };
+
+  pythonImportsCheck = [ "pyscaffoldext.django" ];
+
+  meta = with lib; {
+    description = "Integration of django builtin scaffold cli (django-admin) into PyScaffold";
+    homepage = "https://pypi.org/project/pyscaffoldext-django/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (ehmry) [inline pkgs/development/python-modules/pyscaffoldext-django/default.nix:58]: More Empty maintainers. I won't mark them all.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by MatthewCroughan** [comment]
@ehmry 

```
❯ pwd
/home/matthew/git/nixpkgs/pkgs/development/python-modules

❯ grep -ri 'maintainers; \[ ]' | wc -l
794
```

I think it's fine to leave it this way. I don't think I want to get spammed 16 times by email each time these packages get auto-updated, or asked for permission for their version bumps. 

As for fetchFromGitHub/fetchPypi, it's just a matter of how I initialised the code, but there are even more examples of this also in Nixpkgs, and unless it's going to be made a policy that we can't fetchFromGitHub, I don't think it's worth holding back a merge. Worst case scenario, it simply gets changed and the narHash remains the same.

```
grep -ri 'src = fetchFromGitHub' | wc -l
3595
```

**Review by ghost — APPROVED** [review]

**Pull request merged**